### PR TITLE
Reduce docker cache invalidation for agent targets

### PR DESCRIPTION
**Pull request by jchengsfx**
Reorder some steps in the main Dockerfile to help reduce cache invalidations when building the agent and stage caches.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -250,23 +253,23 @@ RUN find $PYTHONHOME -name "*.pyc" -o -name "*.pyo" | xargs rm
...
-	  vim
-
-RUN apt install -y openjdk-8-jre-headless &&\
-    mkdir -p /opt/root &&\
+        curl \
+        host \
+        iproute2 \
+        netcat \
+        netcat.openbsd \
+        openjdk-8-jre-headless \
```
Comment (keitwb) [inline Dockerfile]: Take the change to this extra-packages stage out, my PR #976 will fix this and also requires some changes to Java in general.

File: Dockerfile [hunk 2]
```diff
@@ -403,10 +407,23 @@ ENV SIGNALFX_BUNDLE_DIR=/bundle \
     LC_ALL=C.UTF-8 \
     LANG=C.UTF-8
 
+RUN curl -fsSL get.docker.com -o /tmp/get-docker.sh &&\
+    sh /tmp/get-docker.sh
+
+# Get integration test deps in here
 RUN pip3 install ipython ipdb
+COPY python/setup.py /tmp/
+RUN pip3 install -e /tmp/
```
Comment (keitwb) [inline Dockerfile]: I am skeptical of this.  In general the developer should be either using the python runner in the bundle dir in the dev image or referencing the packages directly in the `python` directory from the shared mount of the agent code (that is what the `PYTHONPATH=/usr/src/signalfx-agent/python` above accomplishes).  What were you trying to solve here?

Comment (keitwb) [reply Dockerfile]: Ok, I see it is just copied from below.  Still, it doesn't seem right.

File: Dockerfile (hunk 1 above)
Comment (jchengsfx) [reply Dockerfile]: > Take the change to this extra-packages stage out, my PR #976 will fix this and also requires some changes to Java in general.

Will revert and wait for your changes.

File: Dockerfile (hunk 2 above)
Comment (jchengsfx) [reply Dockerfile]: > Ok, I see it is just copied from below. Still, it doesn't seem right.

Yeah, not sure why it was done this way to begin with.  Will change to install using python from the bundle dir instead.

Comment (jchengsfx) [reply Dockerfile]: These lines have been removed.

**Pull request merged**